packages/suite/src/components/suite/OverflowItems.tsx

```diff
@@ -0,0 +1,134 @@
+import { ElementType, ReactNode, useCallback, useEffect, useRef, useState } from 'react';
+import { usePrevious, useShallowCompareEffect, useUpdateEffect } from 'react-use';
+import { useSelector } from 'src/hooks/suite';
+
+type Item = {
+    removeOrder?: number;
+    [key: string]: any;
+};
+
+interface OverflowItemsProps<T> {
+    wrapper?: ElementType;
+    items: T[];
+    visibleItemRenderer: (item: T, index: number) => ReactNode;
+    overflowRenderer: (items: T[]) => ReactNode;
+    minVisibleItems?: number;
+    alwaysRenderOverflow?: boolean;
+}
+
+export function OverflowItems<T extends Item>(props: OverflowItemsProps<T>) {
+    const {
+        items,
+        wrapper: Wrapper = 'div',
+        minVisibleItems = 0,
+        alwaysRenderOverflow = false,
+        visibleItemRenderer,
+        overflowRenderer,
+    } = props;
+
+    const [state, setState] = useState({
+        visibleItems: items,
+        overflowItems: [] as T[],
+    });
+
+    const wrapperRef = useRef<typeof Wrapper>(null);
+
+    const repartition = useCallback(
+        (growing: boolean) => {
+            if (!wrapperRef.current) {
+                return;
+            }
+
+            const wrapper = wrapperRef.current as unknown as Element;
+
+            if (!wrapper) {
+                return;
+            }
+
+            if (growing) {
```

Comment:
So `growing` means that it's reset to all items being visible?

Reply:
Yes, it shows all items and then gradually hide one by one and checking the content width with bounding width

Reply:
then how about instead of `repartition(false)` just use this directly? 
 
```
setState({
    overflowItems: [],
    visibleItems: props.items,
})
                ```
                
Or have another function:

```
const reset = () => setState({
    overflowItems: [],
    visibleItems: props.items,
})
```